packages/endomoat/src/index.js

```diff
@@ -0,0 +1,145 @@
+      let propName = rest.join('.')
+      const itemForBuiltin = policyItem[builtinName]
+      if (typeof itemForBuiltin === 'boolean') {
+        throw new TypeError('Expected a FullAttenuationDefinition; got a boolean')
+      }
+      if (isArray(itemForBuiltin)) {
+        throw new TypeError('Expected a FullAttenuationDefinition; got an array')
+      }
+      const otherParams = itemForBuiltin?.params ?? []
+      policyItem[builtinName] = {
+        attenuate: PROPERTY_ATTENUATOR,
+        params: [...otherParams, propName],
+      }
+    } else {
+      policyItem[key] = value
+    }
+  }
+  return policyItem
+}
+
+/**
+ * @param {Record<string, boolean>} item - A value in `ResourcePolicy`
+ */
+function toEndoRsrcPkgsPolicyPkgs(item) {
+  /** @type {Exclude<import('@endo/compartment-mapper').PackagePolicy['packages'], undefined>} */
+  const policyItem = {}
+  for (const [key, value] of Object.entries(item)) {
+    if (key === LAVAMOAT_PKG_POLICY_ROOT) {
+      // unsure what to do here
+    } else {
+      policyItem[key] = value
+    }
+  }
+  return policyItem
+}
+
+
+/**
+ * @param {Record<string, boolean>} item - A value in `ResourcePolicy`
+ */
+function toEndoRsrcPkgsPolicyGlobals(item) {
+  /** @type {Exclude<import('@endo/compartment-mapper').PackagePolicy['globals'], undefined>} */
+  const policyItem = {}
+  for (const [key, value] of Object.entries(item)) {
+    policyItem[key] = value
+  }
+  return policyItem
+}
+
+/**
+ *
+ * @param {import('lavamoat-core/schema').ResourcePolicy} resources
+ * @returns {import('@endo/compartment-mapper').PackagePolicy<RootPolicy>}
+ */
+function toEndoRsrcPkgsPolicy(resources)  {
+  /** @type {import('@endo/compartment-mapper').PackagePolicy<RootPolicy>['packages']} */
+  let packages
+  /** @type {import('@endo/compartment-mapper').PackagePolicy<RootPolicy>['globals']} */
+  let globals
+  /** @type {import('@endo/compartment-mapper').PackagePolicy<RootPolicy>['builtins']} */
+  let builtins
+
+  if (resources.packages) {
+    packages = toEndoRsrcPkgsPolicyPkgs(resources.packages)
+  }
+
+  if (resources.globals) {
+    globals = toEndoRsrcPkgsPolicyGlobals(resources.globals)
+  }
+
+  if (resources.builtins) {
+    builtins = toEndoRsrcPkgsPolicyBuiltins(resources.builtins)
+  }
+
+  return {packages, globals, builtins}
+
+}
+
+/**
+ *
+ * @param {import('lavamoat-core/schema').LavaMoatPolicySchema} lmPolicy
+ * @returns {LavaMoatEndoPolicy}
+ */
+function toEndoPolicy(lmPolicy) {
+  /** @type {LavaMoatEndoPolicy} */
+  const endoPolicy = {
+    defaultAttenuator: DEFAULT_ATTENUATOR,
+    entry: {
+      [RSRC_POLICY_GLOBALS]: [ENDO_ROOT_POLICY],
+      [RSRC_POLICY_PKGS]: ENDO_WILDCARD_POLICY,
+      [RSRC_POLICY_BUILTINS]: ENDO_WILDCARD_POLICY,
+      noGlobalFreeze: true,
+    },
+    resources: {},
+  }
+
+  if (lmPolicy.resources) {
+    for (const [rsrcName, rsrcPolicy] of Object.entries(lmPolicy.resources)) {
+      if (rsrcName === LAVAMOAT_ROOT_PKG) {
```

Comment:
I don't think we support setting anything on root at this time. Let me know where you've found examples of this :sweat_smile: